doc/manual/src/language/constructs.md

```diff
@@ -1,5 +1,372 @@
 # Language Constructs
 
+## Basic Literals
```

Comment:
I feel like this is unnecessary categorization.
A general introductory statement would be nice though.

```suggestion
The purpose of literals in a language is to turn expressions into new [values](./values.md) in a straightforward way.
<!-- The measure of straightforwardness may differ per syntactic construct, but that's just how the terminology works for programming languages. -->
```

I think one sentence is enough, but I've left a comment for contributors.

Reply:
This is going to get reworked; @fricklerhandwerk asked me to keep the PRs small, so this one is mostly about getting `language/values.md` set up and moving the content that was previously there to roughly the right place before editing it. (This header was previously ‘Primitives’ when this content lived in `values.md`; I changed it to align with the basic values/compound values/functions/externals taxonomy that I'm establishing there now.)